rust/private/rust_analyzer.bzl

```diff
@@ -52,17 +52,21 @@ def _rust_analyzer_aspect_impl(target, ctx):
     if hasattr(ctx.rule.attr, "rustc_flags"):
         cfgs += [f[6:] for f in ctx.rule.attr.rustc_flags if f.startswith("--cfg ") or f.startswith("--cfg=")]
 
-    # Save BuildInfo if we find any (for build script output)
     build_info = None
-    for dep in ctx.rule.attr.deps:
-        if BuildInfo in dep:
-            build_info = dep[BuildInfo]
+    dep_infos = []
+    if hasattr(ctx.rule.attr, "proc_macro_deps"):
```

Comment:
This feels worth a comment - I'm assuming this is for sniffing whether we're a rules_rust rule vs some other one, which happens to also expose `BuildInfo`?

Reply:
This was a bug :) I wanted to test if there is 'deps' attribute, not a 'proc_macro_deps' attribute. Fixed now.